components/script/dom/globalscope.rs

```diff
@@ -2218,6 +2226,16 @@ impl GlobalScope {
         self.evaluate_script_on_global_with_result(code, "", rval, 1)
     }
 
+    unsafe extern "C" fn off_thread_compilation_callback(
+        &self,
+        token: *mut c_void,
+        callbackData: *mut c_void,
+    ) {
+        let castedCallbackData = callbackData as *mut OffThreadCompilationData;
+        let script: JSScript* = FinishOffThreadScript(*(castedCallbackData.context), token);
```

Comment:
To answer your question: here you would use the `OffThreadCompilationData` to then queue a task, and inside that task you would do the rest such as calling `FinishOffThreadScript`. 

Also, the context can be obtained from inside the task, by calling `global.get_cx()` and then you need to de-reference it when needed, for example by doing `FinishOffThreadScript(*global.get_cx(), token);`

(and yes `context` is often used in the other structs I pointed to to refer to the `Trusted<GlobalScope>` but that is just a name to say it's the "context" for the overal task that is being queued. 

So I think this should be enough info to queue the task, and then let me know if you have other questions as to how to handle the `JsScript` further, I would say you should look into calling a method on the global with it.

Reply:
Like other structs this would also not have access to the original context, and needs to be stored. Is that correct?

Reply:
Yes, correct, you need to store the `Trusted<GlobalScope>`(and other things if you need them.